torch/csrc/jit/script/sugared_value.cpp

```diff
@@ -236,6 +242,122 @@ std::shared_ptr<SugaredValue> SimpleValue::call(
   return SugaredValue::call(loc, m, inputs, attributes, n_binders);
 }
 
+Value* SimpleValue::len(const SourceRange& loc, Function& m) {
+  // List, Tuple, Tensor, fill in missing information desugaring
+  Value* val = getValue();
+  TypePtr val_type = val->type();
+  Graph& g = *m.graph();
+  if (val_type->cast<ListType>()) {
+    return g.insert(aten::len, {val}, {}, loc);
+  } else if (val_type->isSubtypeOf(TensorType::get())) {
+    Value* outermost_dim_index = g.insertConstant(0, IntType::get(), loc);
+    // zero-dim tensor error handling
+    Value* num_dim = g.insert(aten::dim, {val}, {}, loc);
```

Comment:
I am not a fan of using so much graph manipulation to write code. I'd prefer this just was a call to a builtin that got the length of the tensor or errored on zero dim.

Reply:
Most of the comments are addressed. For this one, I just followed the old implementation on tensor looping. I thought this a bit, agree with you that we should probably register a builtin prim operator to hide the graph manipulations. Do we have the flexibility of adding prim operators for those graph manipulations?